suite-common/fiat-services/src/rates.ts

```diff
@@ -32,85 +34,118 @@ export const fetchCurrentFiatRates = async (
     ticker: TickerId,
     currency: FiatCurrencyCode,
 ): Promise<number | null | undefined> => {
-    const responseConnect = await scheduleAction(
-        () =>
-            TrezorConnect.blockchainGetCurrentFiatRates({
-                coin: ticker.symbol,
-                token: ticker.tokenAddress,
-                currencies: [currency],
-            }),
-        { timeout: CONNECT_FETCH_TIMEOUT },
-    );
-
-    const rateFromConnect = responseConnect.success
-        ? responseConnect.payload.rates?.[currency]
-        : null;
-
-    if (rateFromConnect) {
-        return rateFromConnect;
+    if (isBlockbookBasedNetwork(ticker.symbol)) {
+        const { success, payload } = await scheduleAction(
+            () =>
+                TrezorConnect.blockchainGetCurrentFiatRates({
+                    coin: ticker.symbol,
+                    token: ticker.tokenAddress,
+                    currencies: [currency],
+                }),
+            { timeout: CONNECT_FETCH_TIMEOUT },
+        );
+
+        const rate = success ? payload.rates?.[currency] : null;
+
+        if (rate) return rate;
+
+        if (!(ticker.symbol === 'btc' && isDesktop)) {
+            // return null because fallbacks doesn't make sense here
+            return null;
+        }
+
+        // Can be electrum for BTC where fallback does make sense
```

Comment:
Don't we have selector to determine if user is using custom BE? It will be better than fallback just directly go to coingecko instead of trying blockbook first.

Reply:
Better detection is for sure good idea but the fallback structure remains the same.

Reply:
2ce974593b8d5ef5049044395c9da6eabad3e369